Compiler/src/tfuncs.jl

```diff
@@ -1033,7 +1033,7 @@ end
         # we may assume that we don't throw
         @assert !boundscheck
         ismod && return false
-        name ⊑ Int || name ⊑ Symbol || return false
+        name ⊑ Int || return false
```

Comment:
Why this change?

Reply:
I was under the impression that `boundscheck=false` was a no-op when `name` is a symbol, since we always throw a `FieldError` if it doesn't exist.  Pre-change:
```
julia> Base.infer_effects(x -> getfield(x, :foo, false), (Pair{String, Bool},))
(+c,+e,+n,+t,+s,+m,+u,+o,+r)

julia> (x -> getfield(x, :foo, false))(Pair("abc", false))
ERROR: type Pair has no field foo
Stacktrace:
 [1] (::var"#15#16")(x::Pair{String, Bool})
   @ Main ./REPL[6]:1
 [2] top-level scope
   @ REPL[6]:1
```

Reply:
Post change:
```
julia> Base.infer_effects(x -> getfield(x, :foo, false), (Pair{String, Bool},))
(+c,+e,!n,+t,+s,+m,!u,+o,+r)

julia> Base.infer_effects(x -> getfield(x, :foo, true), (Pair{String, Bool},))
(+c,+e,!n,+t,+s,+m,+u,+o,+r)

julia> Base.infer_effects(x -> getfield(x, 1, false), (Pair{String, Bool},))
(+c,+e,+n,+t,+s,+m,+u,+o,+r)

julia> Base.infer_effects(x -> getfield(x, 3, false), (Pair{String, Bool},))
(+c,+e,+n,+t,+s,+m,!u,+o,+r)
```